File: app/src/main/java/com/tm/blplayer/network/ApiService.java
package com.tm.blplayer.network;

import com.tm.blplayer.bean.BangumiDetailData;
import com.tm.blplayer.bean.BangumiListData;
import com.tm.blplayer.bean.BaseBean;
import com.tm.blplayer.bean.CategoryListData;
import com.tm.blplayer.bean.HomeData;
import com.tm.blplayer.bean.VideoDetailData;
import com.tm.blplayer.bean.VideoListData;

import io.reactivex.Observable;
import retrofit2.http.GET;
import retrofit2.http.Path;
import retrofit2.http.Query;

/**
 * @author wutongming
 * @description Api接口
 * @since 2017/4/17
 */

public interface ApiService {

    /**
     * 获取首页推荐数据
     */
    @GET("/api/v1/home/")
    Observable<BaseBean<HomeData>> getHome();

    /**
     * 获取视频具体信息
     */
    @GET("/api/v1/video/av{aid}/")
    Observable<BaseBean<VideoDetailData>> getVideoDetail(@Path("aid") String aid);

    /**
     * 获取国创时间表
     */
    @GET("/api/v1/guochuang")
    Observable<BaseBean<BangumiListData>> getGuochuangTimeline();

    /**
     * 获取番剧时间表
     */
    @GET("/api/v1/bangumi")
    Observable<BaseBean<BangumiListData>> getBangumiTimeline();

    /**
     * 获取番剧具体信息
     */
    @GET("/api/v1/bangumi/{sid}/")
    Observable<BaseBean<BangumiDetailData>> getBangumiDetailData(@Path("sid") String sid);

    /**
     * 视频搜索
     */
    @GET("/api/v1/search/")
    Observable<BaseBean<VideoListData>> doSearch(@Query("word") String word, @Query("page") int page, @Query("order") String order);

    /**
     * 分类列表
     */
    @GET("/api/v1/category")
    Observable<BaseBean<CategoryListData>> getCategoryOrder();

    /**
     * 获取一级分类 如动画
     */
    @GET("/api/v1/category/{first_tid}")
    Observable<BaseBean<?>> getCategoryInfo(@Path("first_tid") String firstTid);

    /**
     * 获取二级分类
     */
    @GET("/api/v1/category/{first_tid}/{second_tid}")
    Observable<BaseBean<?>> getCategoryInfo(@Path("first_tid") String firstTid, @Path("second_tid") String secondTid);
}

File: app/src/main/java/com/tm/blplayer/network/ApiService.kt
package com.tm.blplayer.network

import com.tm.blplayer.bean.*
import io.reactivex.Observable
import retrofit2.http.GET
import retrofit2.http.Path
import retrofit2.http.Query

/**
 * @author wutongming
 * *
 * @description Api接口
 * *
 * @since 2017/4/17
 */

interface ApiService {

    /**
     * 获取首页推荐数据
     */
    @get:GET("/api/v1/home/")
    val home: Observable<BaseBean<HomeData>>

    /**
     * 获取视频具体信息
     */
    @GET("/api/v1/video/av{aid}/")
    fun getVideoDetail(@Path("aid") aid: String): Observable<BaseBean<VideoDetailData>>

    /**
     * 获取国创时间表
     */
    @get:GET("/api/v1/guochuang")
    val guochuangTimeline: Observable<BaseBean<BangumiListData>>

    /**
     * 获取番剧时间表
     */
    @get:GET("/api/v1/bangumi")
    val bangumiTimeline: Observable<BaseBean<BangumiListData>>

    /**
     * 获取番剧具体信息
     */
    @GET("/api/v1/bangumi/{sid}/")
    fun getBangumiDetailData(@Path("sid") sid: String): Observable<BaseBean<BangumiDetailData>>

    /**
     * 视频搜索
     */
    @GET("/api/v1/search/")
    fun doSearch(@Query("word") word: String, @Query("page") page: Int, @Query("order") order: String): Observable<BaseBean<VideoListData>>

    /**
     * 分类列表
     */
    @get:GET("/api/v1/category")
    val categoryOrder: Observable<BaseBean<CategoryListData>>

    /**
     * 获取一级分类 如动画
     */
    @GET("/api/v1/category/{first_tid}")
    fun getCategoryInfo(@Path("first_tid") firstTid: String): Observable<BaseBean<*>>

    /**
     * 获取二级分类
     */
    @GET("/api/v1/category/{first_tid}/{second_tid}")
    fun getCategoryInfo(@Path("first_tid") firstTid: String, @Path("second_tid") secondTid: String): Observable<BaseBean<*>>
}
